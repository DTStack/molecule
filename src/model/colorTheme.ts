export interface IColors {
    [colorId: string]: string;
}
export interface TokenColor extends Object {
    name?: string;
    scope?: string | string[];
    settings?: object;
}

/**
 * Color scheme used by the OS and by color themes.
 */
export enum ColorScheme {
    DARK = 'dark',
    LIGHT = 'light',
    HIGH_CONTRAST = 'hc',
}

export enum ColorThemeMode {
    dark = 'dark',
    light = 'light',
}

export enum ColorThemeEvent {
    onChange = 'colorTheme.onChange',
}

export interface IColorTheme {
    /**
     * The id of component, theme will be applied by this ID
     */
    id: string;
    label: string;
    name?: string;
    uiTheme?: string;
    path?: string;
    description?: string;
    type?: ColorScheme;
    colors?: IColors;
    tokenColors?: TokenColor[];
    /**
     * The semanticTokenColors mappings as well as
     * the semanticHighlighting setting
     * allow to enhance the highlighting in the editor
     * More info visit: https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide
     */
    semanticHighlighting?: boolean;
}
