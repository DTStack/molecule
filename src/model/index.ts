export * from './workbench';
export * from './extension';
export * from './iconTheme';
export * from './settings';
export * from './notification';
export * from './problems';
export * from './colorTheme';
export * from './keybinding';
