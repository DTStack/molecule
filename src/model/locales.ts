export interface ILocales {
    /**
     * The id of component, locale will be applied by this ID
     */
    id: string;
    path: string;
}
