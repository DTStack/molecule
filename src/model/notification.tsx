import { UniqueId } from 'mo/common/types';
import { IActionBarItemProps } from 'mo/components/actionBar';
import React from 'react';
import type { IStatusBarItem } from './workbench/statusBar';

export enum NotificationStatus {
    Read = 1,
    WaitRead = 2,
}

export interface INotificationItem<T = any> {
    id: UniqueId;
    value: T;
    render?(item: INotificationItem): React.ReactNode;
    status?: NotificationStatus;
}

export interface INotification<T = any>
    extends IStatusBarItem<INotificationItem<T>[]> {
    showNotifications?: boolean;
    actionBar?: IActionBarItemProps[];
}

export class NotificationModel<T> implements INotification<T> {
    public id: UniqueId;
    public name: string;
    public data: INotificationItem<T>[];
    public sortIndex: number;
    public render: () => React.ReactNode;
    public showNotifications: boolean;
    public actionBar: IActionBarItemProps[];

    constructor(
        id: UniqueId = '',
        name: string = '',
        data: INotificationItem<T>[] = [],
        sortIndex: number = 1,
        showNotifications: boolean = false,
        actionBar: IActionBarItemProps[] = [],
        render: () => React.ReactNode
    ) {
        this.id = id;
        this.name = name;
        this.sortIndex = sortIndex;
        this.render = render;
        this.showNotifications = showNotifications;
        this.data = data;
        this.actionBar = actionBar;
    }
}
