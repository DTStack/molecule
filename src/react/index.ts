export * from './component';
export * from './helper';
export * from './connector';
export * from './controller';
