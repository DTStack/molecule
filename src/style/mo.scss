@import 'mo/style/common';
@import 'mo/style/animation';

// ============= The Workbench parts =============
@import 'mo/components/actionBar/style';
@import 'mo/components/breadcrumb/style';
@import 'mo/components/button/style';
@import 'mo/components/checkbox/style';
@import 'mo/components/collapse/style';
@import 'mo/components/contextView/style';
@import 'mo/components/dialog/style';
@import 'mo/components/dropdown/style';
@import 'mo/components/input/style';
@import 'mo/components/list/style';
@import 'mo/components/menu/style';
@import 'mo/components/scrollable/style';
@import 'mo/components/select/style';
@import 'mo/components/tabs/style';
@import 'mo/components/toolbar/style';
@import 'mo/components/tree/style';
@import 'mo/components/search/style';
@import 'mo/components/tooltip/style';
@import 'mo/components/split/style';

// ============= The Workbench core component =============
@import 'mo/workbench/style';
@import 'mo/workbench/activityBar/style';
@import 'mo/workbench/editor/style';
@import 'mo/workbench/editor/welcome/style';
@import 'mo/workbench/menuBar/style';
@import 'mo/workbench/panel/style';
@import 'mo/workbench/settings/style';
@import 'mo/workbench/sidebar/style';
@import 'mo/workbench/statusBar/style';
@import 'mo/workbench/notification/style';
@import 'mo/workbench/sidebar/explore/style';
@import 'mo/workbench/problems/style';
@import 'mo/workbench/notification/statusBarView/style';

.#{$prefix} {
    bottom: 0;
    box-sizing: border-box;
    font-size: 13px;
    height: 100%;
    left: 0;
    margin: 0;
    outline: 0;
    overflow: hidden;
    padding: 0;
    position: absolute;
    right: 0;
    top: 0;
    user-select: none;
    width: 100%;
}

.mac {
    font-family: -apple-system, BlinkMacSystemFont, system-ui, sans-serif;
}

.center {
    text-align: center;
}

.hidden {
    display: none;
}
