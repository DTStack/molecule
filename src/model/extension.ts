import type { UniqueId } from 'mo/common/types';
import { IExtensionService } from 'mo/services';
import { IColorTheme } from './colorTheme';
import { IIconTheme } from './iconTheme';
import { ILocale } from 'mo/i18n';

/**
 * Defines extension types
 */
export enum IExtensionType {
    Theme = 'Themes',
    Normal = 'normal',
    Settings = 'settings',
    Locals = 'locales',
    Menus = 'menus',
    Workbench = 'workbench',
}

export enum IContributeType {
    Languages = 'languages',
    Commands = 'commands',
    Configuration = 'configuration',
    Grammar = 'grammars',
    Themes = 'themes',
    IconTheme = 'iconThemes',
}

export interface IContribute {
    [IContributeType.Languages]?: ILocale[];
    [IContributeType.Commands]?: any;
    [IContributeType.Configuration]?: any;
    [IContributeType.Grammar]?: any;
    [IContributeType.Themes]?: IColorTheme[];
    [IContributeType.IconTheme]?: IIconTheme[];
}

/**
 * The interface of extension,
 * there need every extension to implement this interface
 */
export interface IExtension {
    /**
     * The ID of extension required
     */
    id: UniqueId;
    /**
     * The name of extension
     */
    name: string;
    /**
     * The display name of extension
     */
    displayName?: string;
    /**
     * The version of extension
     */
    version?: string;
    /**
     * The categories of extension
     */
    categories?: IExtensionType[];
    /**
     * The kind of extension
     */
    extensionKind?: IExtensionType[];
    /**
     * The main file path of extension
     * Extension system will load the extension by this file
     */
    contributes?: IContribute;
    /**
     * The entry of extension
     */
    main?: string;
    /**
     * The Icon of extension
     */
    icon?: string | JSX.Element;
    /**
     * The description of extension
     */
    description?: string;
    /**
     * The publisher of extension
     */
    publisher?: string;
    /**
     * The path of extension
     */
    path?: string;
    /**
     * Whether disable current extension, the extension default status is enable
     */
    disable?: boolean;
    /**
     * Do something you want when the Extension is activating.
     * The ExtensionService will call the `activate` method after
     * added the Extension instance.
     * @param extensionCtx The Context of Extension instance
     */
    activate(extensionCtx: IExtensionService): void;
    /**
     * Do something when the Extension disposing.
     * For example, you can recover the UI state, or remove the Objects in memory.
     * @param extensionCtx The Context of Extension instance
     */
    dispose(extensionCtx: IExtensionService): void;
}
