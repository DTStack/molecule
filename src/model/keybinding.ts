import { KeyCode } from 'mo/monaco';

export const KeyCodeString: Partial<{ [key in KeyCode]: string }> = {
    [KeyCode.Unknown]: '',
    [KeyCode.Backspace]: '⌫',
    [KeyCode.Tab]: '⇥',
    [KeyCode.Enter]: '↩',
    [KeyCode.PageUp]: '↑',
    [KeyCode.PageDown]: '↓',
    [KeyCode.Digit0]: '0',
    [KeyCode.Digit1]: '1',
    [KeyCode.Digit2]: '2',
    [KeyCode.Digit3]: '3',
    [KeyCode.Digit4]: '4',
    [KeyCode.Digit5]: '5',
    [KeyCode.Digit6]: '6',
    [KeyCode.Digit7]: '7',
    [KeyCode.Digit8]: '8',
    [KeyCode.Digit9]: '9',
    [KeyCode.KeyA]: 'A',
    [KeyCode.KeyB]: 'B',
    [KeyCode.KeyC]: 'C',
    [KeyCode.KeyD]: 'D',
    [KeyCode.KeyE]: 'E',
    [KeyCode.KeyF]: 'F',
    [KeyCode.KeyG]: 'G',
    [KeyCode.KeyH]: 'H',
    [KeyCode.KeyI]: 'I',
    [KeyCode.KeyJ]: 'J',
    [KeyCode.KeyK]: 'K',
    [KeyCode.KeyL]: 'L',
    [KeyCode.KeyM]: 'M',
    [KeyCode.KeyN]: 'N',
    [KeyCode.KeyO]: 'O',
    [KeyCode.KeyP]: 'P',
    [KeyCode.KeyQ]: 'Q',
    [KeyCode.KeyR]: 'R',
    [KeyCode.KeyS]: 'S',
    [KeyCode.KeyT]: 'T',
    [KeyCode.KeyU]: 'U',
    [KeyCode.KeyV]: 'V',
    [KeyCode.KeyW]: 'W',
    [KeyCode.KeyX]: 'X',
    [KeyCode.KeyY]: 'Y',
    [KeyCode.KeyZ]: 'Z',
    [KeyCode.Semicolon]: ';',
    [KeyCode.Equal]: '+',
    [KeyCode.Comma]: ',',
    [KeyCode.Minus]: '-',
    [KeyCode.Period]: '.',
    [KeyCode.Slash]: '/',
    [KeyCode.Backquote]: '~',
    [KeyCode.BracketLeft]: '[',
    [KeyCode.Backslash]: '\\',
    [KeyCode.BracketRight]: ']',
    [KeyCode.Quote]: '"',
};

export interface ISimpleKeybinding {
    ctrlKey: boolean;
    shiftKey: boolean;
    altKey: boolean;
    metaKey: boolean;
    keyCode: KeyCode;
}
