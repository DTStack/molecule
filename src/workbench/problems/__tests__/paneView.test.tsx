import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import ProblemsPaneView from '../paneView';
import { MarkerSeverity } from 'mo/model';

const mockErrorPro = {
    id: 1,
    name: '0-1',
    value: {
        code: 'endLineNumber',
        message: '语法错误',
        startLineNumber: 0,
        startColumn: 1,
        endLineNumber: 0,
        endColumn: 1,
        status: MarkerSeverity.Error,
    },
    children: [],
};
const mockWarningPro = {
    id: 2,
    name: '0-2',
    value: {
        code: 'endLineNumber',
        message: '解析错误',
        startLineNumber: 0,
        startColumn: 1,
        endLineNumber: 0,
        endColumn: 1,
        status: MarkerSeverity.Warning,
    },
    children: [],
};

const mockInfoPro = {
    id: 3,
    name: '0-3',
    value: {
        code: 'endLineNumber',
        message: '有点问题',
        startLineNumber: 0,
        startColumn: 1,
        endLineNumber: 0,
        endColumn: 1,
        status: MarkerSeverity.Info,
    },
    children: [],
};

const mockDefaultPro = {
    id: 4,
    name: '0-4',
    value: {
        code: 'endLineNumber',
        message: 'problems',
        startLineNumber: 0,
        startColumn: 1,
        endLineNumber: 0,
        endColumn: 1,
        status: 0,
    },
    children: [],
};
const mockProblemFile = {
    id: 1,
    name: 'text.tsx',
    value: {
        code: 'text.tsx',
        message: '文件夹',
        startLineNumber: 0,
        startColumn: 1,
        endLineNumber: 0,
        endColumn: 1,
        status: 1,
    },
    children: [mockErrorPro, mockWarningPro, mockInfoPro, mockDefaultPro],
};

const mockRootData = {
    id: 'test',
    name: 'test-name',
    data: [mockProblemFile],
};

describe('The PaneView Component', () => {
    afterEach(cleanup);

    test('Match snapshot', () => {
        const component = create(<ProblemsPaneView {...mockRootData} />, {
            createNodeMock: () => {
                const dom = document.createElement('div');
                return dom;
            },
        });
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Should render no data tips in Pane', () => {
        const { getByText } = render(
            <ProblemsPaneView id="test" name="problem" data={[]} />
        );

        const tips = getByText('未在工作区检测到问题');
        expect(tips).toBeInTheDocument();
    });
});
