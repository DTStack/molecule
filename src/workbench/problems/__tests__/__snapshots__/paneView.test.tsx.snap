// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PaneView Component Match snapshot 1`] = `
<DocumentFragment>
  <div
    class="mo-problems"
  >
    <div
      class="mo-tree mo-problems__treeview"
      draggable="false"
      role="tree"
    >
      <div
        class="mo-tree__treenode mo-tree__treenode--active mo-tree__treenode--open"
        data-id="mo_treeNode_1"
        data-indent="0"
        data-key="1"
        draggable="false"
        tabindex="0"
      >
        <div
          class="mo-tree__indent"
          style="width: 0px;"
        />
        <span
          class="codicon codicon-chevron-down"
        />
        <span
          class="mo-tree__treenode__title"
        >
          <span
            class="mo-problems__treeview__treeNode"
          >
            text.tsx
            <span
              class="mo-problems__treeview__treeNode__badge"
            >
              4
            </span>
          </span>
        </span>
      </div>
      <div
        class="mo-tree__treenode mo-tree__treenode--active mo-tree__treenode--open"
        data-id="mo_treeNode_1_1"
        data-indent="1"
        data-key="1"
        draggable="false"
        tabindex="0"
      >
        <div
          class="mo-tree__indent"
          style="width: 8px;"
        >
          <div
            class="mo-tree__indent__guide"
          />
        </div>
        <span
          class="mo-tree__treenode__title"
        >
          <span
            class="mo-problems__treeview__treeLeaf"
          >
            <span
              class="codicon codicon-error"
            />
            <span>
              语法错误
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              endLineNumber
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              [0, 1]
            </span>
          </span>
        </span>
      </div>
      <div
        class="mo-tree__treenode mo-tree__treenode--close"
        data-id="mo_treeNode_1_2"
        data-indent="1"
        data-key="2"
        draggable="false"
        tabindex="0"
      >
        <div
          class="mo-tree__indent"
          style="width: 8px;"
        >
          <div
            class="mo-tree__indent__guide"
          />
        </div>
        <span
          class="mo-tree__treenode__title"
        >
          <span
            class="mo-problems__treeview__treeLeaf"
          >
            <span
              class="codicon codicon-warning"
            />
            <span>
              解析错误
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              endLineNumber
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              [0, 1]
            </span>
          </span>
        </span>
      </div>
      <div
        class="mo-tree__treenode mo-tree__treenode--close"
        data-id="mo_treeNode_1_3"
        data-indent="1"
        data-key="3"
        draggable="false"
        tabindex="0"
      >
        <div
          class="mo-tree__indent"
          style="width: 8px;"
        >
          <div
            class="mo-tree__indent__guide"
          />
        </div>
        <span
          class="mo-tree__treenode__title"
        >
          <span
            class="mo-problems__treeview__treeLeaf"
          >
            <span
              class="codicon codicon-info"
            />
            <span>
              有点问题
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              endLineNumber
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              [0, 1]
            </span>
          </span>
        </span>
      </div>
      <div
        class="mo-tree__treenode mo-tree__treenode--close"
        data-id="mo_treeNode_1_4"
        data-indent="1"
        data-key="4"
        draggable="false"
        tabindex="0"
      >
        <div
          class="mo-tree__indent"
          style="width: 8px;"
        >
          <div
            class="mo-tree__indent__guide"
          />
        </div>
        <span
          class="mo-tree__treenode__title"
        >
          <span
            class="mo-problems__treeview__treeLeaf"
          >
            <span>
              problems
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              endLineNumber
            </span>
            <span
              class="mo-problems__treeview__treeLeaf__subInfo"
            >
              [0, 1]
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
